File: 238-product-of-array-except-self/238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans(nums.size(),1);
        int num = 1;
        
        for(int i = 0; i < nums.size(); i++) {
            ans[i] = num;
            num *= nums[i];
        }
        
        num = 1;
        
        for(int i = nums.size() -1; i >= 0; i--) {
            ans[i] *= num;
            num *= nums[i];
        }
        
        return ans;
    }
};

File: 238-product-of-array-except-self/238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [1] * len(nums)
        
        prefix = 1
        
        for idx in range(len(nums)):
            ans[idx] = prefix
            prefix *= nums[idx]
        
        postfix = 1
        
        for idx in range(len(nums)-1,-1,-1):
            ans[idx] *= postfix
            postfix *= nums[idx]
        
        
        return ans